pex/commands/bdist_pex.py

```diff
@@ -41,67 +46,80 @@ def _write(self, pex_builder, target, script=None):
-    parser, options_builder = configure_clp()
-    package_dir = os.path.dirname(os.path.realpath(os.path.expanduser(
-        self.distribution.script_name)))
 
+    package_dir = os.path.dirname(os.path.realpath(os.path.expanduser(
+      self.distribution.script_name)))
     if self.bdist_dir is None:
       self.bdist_dir = os.path.join(package_dir, 'dist')
 
-    options, reqs = parser.parse_args(self.pex_args)
+    console_scripts = self.parse_entry_points()
 
-    # Update cache_dir with pex_root in case this is being called directly.
-    if options.cache_dir:
-      options.cache_dir = make_relative_to_root(options.cache_dir)
-    options.interpreter_cache_dir = make_relative_to_root(options.interpreter_cache_dir)
+    pex_specs = []
+    if self.bdist_all:
+      # Write all entry points into unversioned pex files.
+      pex_specs.extend((script_name, os.path.join(self.bdist_dir, script_name))
+                       for script_name in console_scripts)
+    else:
+      target = os.path.join(self.bdist_dir, name + '-' + version + '.pex')
+      pex_specs.append((name if name in console_scripts else None, target))
 
-    if options.entry_point or options.script:
-      die('Must not specify entry_point or script to --pex-args')
+    # In order for code to run to here, pex is on the sys.path - make sure to propagate the
+    # sys.path so the subprocess can find us.
+    env = os.environ.copy()
+    env['PYTHONPATH'] = os.pathsep.join(sys.path)
 
-    reqs = [package_dir] + reqs
+    args = [sys.executable, '-s', '-m', 'pex.bin.pex', package_dir] + self.pex_args
 
-    with ENV.patch(PEX_VERBOSE=str(options.verbosity), PEX_ROOT=options.pex_root):
-      pex_builder = build_pex(reqs, options, options_builder)
+    # A hack to get the existing distutils logging level.
+    existing_level = log.set_threshold(log.INFO)
+    log.set_threshold(existing_level)
+    if existing_level < log.INFO and options.verbosity == 0:
+      args.append('-v')
 
-    console_scripts = self.parse_entry_points()
-
-    target = os.path.join(self.bdist_dir, name + '-' + version + '.pex')
-    if self.bdist_all:
-      # Write all entry points into unversioned pex files.
-      for script_name in console_scripts:
-        target = os.path.join(self.bdist_dir, script_name)
+    for script_name, target in pex_specs:
+      cmd = args + ['-o', target]
+      if script_name:
         log.info('Writing %s to %s' % (script_name, target))
-        self._write(pex_builder, target, script=script_name)
-    elif name in console_scripts:
-      # The package has a namesake entry point, so use it.
-      log.info('Writing %s to %s' % (name, target))
-      self._write(pex_builder, target, script=name)
-    else:
-      # The package has no namesake entry point, so build an environment pex.
-      log.info('Writing environment pex into %s' % target)
-      self._write(pex_builder, target, script=None)
+        cmd += ['-c', script_name]
```

Comment:
Are there long argument names for these, like `--compile` instead of `-c`?

Reply:
Lengthened